tarantool-driver/src/main/java/com/sopovs/moradanen/tarantool/TarantoolPooledClientSource.java

```diff
@@ -111,6 +113,10 @@ public void close() {
                 if (poolClosed) {
                     client.close();
                 } else {
+                    if (lastResult != null && lastResult.hasNext()) {
```

Comment:
Thus no good.
Becuase last resul could be from another client. For example

Thread A
          execute                          close
------------*-----------------------*-------------
Tread B
  execute                 close
--- *------------------*---------------


In this case close in Thread B will close result from ThreadA. And we have two errors. the First is Thread A will lost some results and will fail. And Client from Thread B will returned to the pool in wrong state,

Reply:
`lastResult` is the field of TarantoolClientProxy class - it is bound not even to particular client, but to single lease of that client from the pool.

Reply:
Yes I understand it. And thats is it would map to Exception.
1. Thread A. Call execute(). ClientA get from pool. LastResult in ThreadPool = ResultA
2. Thread B. Call execute(). ClientB get from pool. LastResult in ThreadPool = ResultB
3. Thread A. Call client.close(). In this time ThreadPool client closed LastResult which is equal ResulB. ClientA return to the pool. If user doesn't close resultA muanually then we have Client A is invalid state.
3.5 While reading data from ResultB we will get an exception because ResultB is closed by Pool.

Reply:
I was wrong. It's good.
I thought that TarantoolPooledClient is singleton.